### DISCUSSION TITLE
Build configuration names without spaces

### DISCUSSION THREAD

**Discussion by namiestew** [premake/premake-core#2058] [General]
When using 'configurations' with 'platforms', the resulting build configuration name ends up with whitespace between two words. For example, "Debug Win32" as shown here:

![image](https://user-images.githubusercontent.com/102791593/231425193-a8558b46-1903-4a28-b1c3-fc99c0ce1ea8.png)

Is there any way I can generate build configuration names without spaces when using 'configurations' with 'platforms'? The whitespace in the name causes build issues when compiling for Linux with Ninja.

**Comment by samsinsane** [comment]
Premake-core doesn't have any Ninja support. Can you provide any further context to your issue?

**Reply by namiestew** [reply]
Hi, in a Visual Studio Linux MSBuild Project when Ninja is enabled for incremental build, the spaces in the build configuration names will cause a compilation error. However, this isn't about support for Ninja... I am just asking if there is any way I can set premake to not produce build configuration names with spaces.

**Reply by samsinsane** [reply]
I'm not seeing any spaces, I can see `Debug|Win32` but not `Debug Win32`. Can you provide the error, the specific part of the .sln or .vcxproj that has the space, and/or a minimum repro case?

**Reply by namiestew** [reply]
The screenshot in the initial post taken from premake [documentation](https://premake.github.io/docs/Configurations-and-Platforms) already clearly shows how the configuration names will end up with spaces. And I have already said **my issue is not with the error**. 

I asked
> Is there any way I can generate build configuration names without spaces when using 'configurations' with 'platforms'?

If this is not possible, may I know where exactly in the premake source code do these two values get concatenated so I can modify it myself?

**Reply by samsinsane** [reply]
> Is there any way I can generate build configuration names without spaces when using 'configurations' with 'platforms'?

No, because they aren't separated by spaces.
